edb/ir/staeval.py

```diff
@@ -422,10 +422,14 @@ def cast_const_to_python(ir: irast.TypeCast, schema: s_schema.Schema) -> Any:
     sval = evaluate_to_python_val(ir.expr, schema=schema)
     if sval is None:
         return None
-    elif isinstance(sval, tuple):
-        return tuple(pytype(elem) for elem in sval)
+    if pytype is bool and isinstance(sval, str):
+        conv = lambda v: v.lower() == 'true'
```

Comment:
This should really replicate `StrToBool` and raise an exception if the value is neither 'true' nor 'false'.

Reply:
Right, I've also seen the same. Will fix!